syft/workers/base.py

```diff
@@ -714,6 +714,11 @@ def search(self, *query):
         for key, tensor in self._objects.items():
             found_something = True
             for query_item in query:
+                # If deserialization produced a bytes object instead of a string,
```

Comment:
good catch here - i wonder if we could handle this more generically in serde.py - seems like we should

Reply:
Actually, yeah, this makes a lot of sense. I need to study a bit the serde module to see how to change it to handle these cases.

Reply:
Seems like msgpack should support that through a couple of args to the loads/dumps methods as described here: https://github.com/msgpack/msgpack-python#string-and-binary-type
but when I set these args a bunch of tests fails so I'll have to look into them. Also, I notice now that in this file there is already one other instance where we check for a `bytes` object. Looks like someone has already faced this issue.